rpc/cloudagent/compute/availabilityset/moc_cloudagent_availabilityset.proto

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/compute";
+package moc.cloudagent.compute;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+
+message AvailabilitySetRequest {
+	repeated AvailabilitySet AvailabilitySets = 1;
+	Operation OperationType = 2;
+}
+
+message AvailabilitySetResponse {
+	repeated AvailabilitySet AvailabilitySets = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message VirtualMachineReference {
+	string groupName = 1;
+	string name = 2;
+	string realizedName = 3;
+}
+
+// avset structure is a flattened version of the model in the Azure sdk for go at
+// https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/compute/armcompute/models.go
+message AvailabilitySet {
+    string name = 1;
+    string id = 2;
+    string locationName = 3;
+    string groupName = 4;
+    Status status = 5;
+    Tags tags = 6;
+	int32 platformFaultDomainCount = 7;
+	repeated VirtualMachineReference virtualMachines = 8;
```

Comment:
Having bidirectional representation is easier for validation. We need to make sure there is only one source of truth and the other reference is based of the source of truth. We should not have two brains updating the references. This could cause many race conditions

Reply:
Not a change request. Just to call out what to watch out for

Reply:
We don't have 2 brains.  The source of truth for VM membership is the VM provider.